src/alda/repl/commands/ReplCommand.java

```diff
@@ -13,16 +14,18 @@
  */
 public interface ReplCommand {
 
-
   /**
    * Runs this ReplCommand given the selected history.
    * This ReplCommand will modify the StringBuffer passed in to do it's actions.
    * @param args The arguments to this repl command
    * @param history The current history string for this repl
    * @param server The server to play pull data from, if needed.
    * @param reader A consoleReader so this command can prompt for user input
+   * @param newInstrument A function to call if the current instrument changes.
+   *   The new current instrument will be passed in as a string.
    */
-  public void act(String args, StringBuffer history, AldaServer server, ConsoleReader reader);
+  public void act(String args, StringBuffer history, AldaServer server,
+                  ConsoleReader reader, Consumer<String> newInstrument);
```

Comment:
Yup, this is a good idea! (and it would reduce the shared json parsing across the individual scores).